libs/wingsdk/src/utils/resource-names.ts

```diff
@@ -0,0 +1,38 @@
+export enum CaseConventions {
+  LOWERCASE = "lowercase",
+  UPPERCASE = "uppercase",
+}
+
+export interface NameOptions {
+  readonly resource?: string;
+  readonly regexMatch: RegExp;
+  readonly charReplacer: string;
+  readonly maxLen: number;
+  readonly minLen?: number;
+  readonly case?: CaseConventions;
+  readonly avoidStartWith?: string;
+  readonly prefix?: string;
+}
+
+export class ResourceNames {
+  public static of(name: string, props: NameOptions): string {
```

Comment:
Let's give this method another name, like `generateName` or `makeName`. `of` is typically reserved for methods that find a resource in some `constructs` scope (like `Stack.of(scope)` or `App.of(scope)`).